[2025-08-07T01:30:45.665+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-08-07T01:30:45.696+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: register_high_latency_model.rollback_model manual__2025-08-07T01:30:40.249720+00:00 [queued]>
[2025-08-07T01:30:45.705+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: register_high_latency_model.rollback_model manual__2025-08-07T01:30:40.249720+00:00 [queued]>
[2025-08-07T01:30:45.705+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2025-08-07T01:30:45.720+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): rollback_model> on 2025-08-07 01:30:40.249720+00:00
[2025-08-07T01:30:45.732+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=2523) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-08-07T01:30:45.734+0000] {standard_task_runner.py:63} INFO - Started process 2533 to run task
[2025-08-07T01:30:45.734+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'register_high_latency_model', 'rollback_model', 'manual__2025-08-07T01:30:40.249720+00:00', '--job-id', '11', '--raw', '--subdir', 'DAGS_FOLDER/test_model_rollback.py', '--cfg-path', '/tmp/tmpcm8fl56z']
[2025-08-07T01:30:45.738+0000] {standard_task_runner.py:91} INFO - Job 11: Subtask rollback_model
[2025-08-07T01:30:45.794+0000] {task_command.py:426} INFO - Running <TaskInstance: register_high_latency_model.rollback_model manual__2025-08-07T01:30:40.249720+00:00 [running]> on host 875a2765ee06
[2025-08-07T01:30:45.909+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='register_high_latency_model' AIRFLOW_CTX_TASK_ID='rollback_model' AIRFLOW_CTX_EXECUTION_DATE='2025-08-07T01:30:40.249720+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-08-07T01:30:40.249720+00:00'
[2025-08-07T01:30:45.911+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-08-07T01:30:46.026+0000] {logging_mixin.py:188} INFO - üì• Downloading latest production model from registry...
[2025-08-07T01:30:46.028+0000] {warnings.py:110} WARNING - /opt/***/dags/test_model_rollback.py:51: FutureWarning: ``mlflow.tracking.client.MlflowClient.get_latest_versions`` is deprecated since 2.9.0. Model registry stages will be removed in a future major release. To learn more about the deprecation of model registry stages, see our migration guide here: https://mlflow.org/docs/latest/model-registry.html#migrating-from-stages
  latest_production = client.get_latest_versions(MODEL_NAME, stages=["Production"])[0]

[2025-08-07T01:30:46.568+0000] {credentials.py:1147} INFO - Found credentials in environment variables.
[2025-08-07T01:30:46.880+0000] {logging_mixin.py:188} WARNING - Downloading artifacts:   0%|          | 0/5 [00:00<?, ?it/s]
[2025-08-07T01:30:46.881+0000] {logging_mixin.py:188} WARNING - Downloading artifacts:  20%|##        | 1/5 [00:00<00:00, 5637.51it/s]
[2025-08-07T01:30:46.882+0000] {logging_mixin.py:188} WARNING - Downloading artifacts:  40%|####      | 2/5 [00:00<00:00, 2023.30it/s]
[2025-08-07T01:30:46.883+0000] {logging_mixin.py:188} WARNING - Downloading artifacts:  60%|######    | 3/5 [00:00<00:00, 1764.29it/s]
[2025-08-07T01:30:46.883+0000] {logging_mixin.py:188} WARNING - Downloading artifacts:  80%|########  | 4/5 [00:00<00:00, 1684.29it/s]
[2025-08-07T01:30:46.884+0000] {logging_mixin.py:188} WARNING - Downloading artifacts: 100%|##########| 5/5 [00:00<00:00, 1611.46it/s]
[2025-08-07T01:30:46.885+0000] {logging_mixin.py:188} WARNING - Downloading artifacts: 100%|##########| 5/5 [00:00<00:00, 1297.98it/s]
[2025-08-07T01:30:46.969+0000] {logging_mixin.py:188} INFO - Constructing slow inference model...
[2025-08-07T01:30:46.970+0000] {logging_mixin.py:188} INFO - üì¶ Model prepared, registering with MLflow...
[2025-08-07T01:30:47.100+0000] {logging_mixin.py:188} WARNING - 2025/08/07 01:30:47 WARNING mlflow.utils.git_utils: Failed to import Git (the Git executable is probably not on your PATH), so Git SHA is not available. Error: Failed to initialize: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh(<full-path-to-git-executable>)

All git commands will error until this is rectified.

This initial message can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|silent|none|n|0: for no message or exception
    - warn|w|warning|log|l|1: for a warning message (logging level CRITICAL, displayed by default)
    - error|e|exception|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet
[2025-08-07T01:30:47.127+0000] {logging_mixin.py:188} WARNING - 2025/08/07 01:30:47 WARNING mlflow.models.model: `artifact_path` is deprecated. Please use `name` instead.
[2025-08-07T01:30:54.431+0000] {logging_mixin.py:188} WARNING - 2025/08/07 01:30:54 WARNING mlflow.utils.environment: Encountered an unexpected error while inferring pip requirements (model URI: /tmp/tmp_jslrndj/model/model.pkl, flavor: sklearn). Fall back to return ['scikit-learn==1.7.1', 'cloudpickle==3.1.1']. Set logging level to DEBUG to see the full traceback.
[2025-08-07T01:30:54.444+0000] {logging_mixin.py:188} WARNING - 2025/08/07 01:30:54 WARNING mlflow.models.model: Model logged without a signature and input example. Please set `input_example` parameter when logging the model to auto infer the model signature.
[2025-08-07T01:30:54.689+0000] {logging_mixin.py:188} WARNING - Registered model 'FraudDetectionModel' already exists. Creating a new version of this model...
[2025-08-07T01:30:54.722+0000] {logging_mixin.py:188} WARNING - 2025/08/07 01:30:54 INFO mlflow.store.model_registry.abstract_store: Waiting up to 300 seconds for model version to finish creation. Model name: FraudDetectionModel, version 2
[2025-08-07T01:30:54.723+0000] {logging_mixin.py:188} WARNING - Created version '2' of model 'FraudDetectionModel'.
[2025-08-07T01:30:54.827+0000] {logging_mixin.py:188} INFO - üèÉ View run skittish-lamb-949 at: http://mlflow:5000/#/experiments/1/runs/9bdaea3b262d4aae9f268bad357324e1
[2025-08-07T01:30:54.828+0000] {logging_mixin.py:188} INFO - üß™ View experiment at: http://mlflow:5000/#/experiments/1
[2025-08-07T01:30:54.853+0000] {logging_mixin.py:188} INFO - ‚úÖ High latency model registered successfully.
[2025-08-07T01:30:54.854+0000] {warnings.py:110} WARNING - /opt/***/dags/test_model_rollback.py:86: FutureWarning: ``mlflow.tracking.client.MlflowClient.get_latest_versions`` is deprecated since 2.9.0. Model registry stages will be removed in a future major release. To learn more about the deprecation of model registry stages, see our migration guide here: https://mlflow.org/docs/latest/model-registry.html#migrating-from-stages
  model_version = client.get_latest_versions("FraudDetectionModel", stages=["None"])[-1]

[2025-08-07T01:30:55.043+0000] {warnings.py:110} WARNING - /opt/***/dags/test_model_rollback.py:87: FutureWarning: ``mlflow.tracking.client.MlflowClient.transition_model_version_stage`` is deprecated since 2.9.0. Model registry stages will be removed in a future major release. To learn more about the deprecation of model registry stages, see our migration guide here: https://mlflow.org/docs/latest/model-registry.html#migrating-from-stages
  client.transition_model_version_stage(

[2025-08-07T01:30:55.091+0000] {logging_mixin.py:188} INFO - üöÄ High latency model version 2 deployed to Production!
[2025-08-07T01:30:55.092+0000] {python.py:237} INFO - Done. Returned value was: None
[2025-08-07T01:30:55.093+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-08-07T01:30:55.106+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=register_high_latency_model, task_id=rollback_model, run_id=manual__2025-08-07T01:30:40.249720+00:00, execution_date=20250807T013040, start_date=20250807T013045, end_date=20250807T013055
[2025-08-07T01:30:55.150+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-08-07T01:30:55.172+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-08-07T01:30:55.175+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
